### ISSUE TITLE
Undefined field qid

### ISSUE THREAD

**Issue by ansell** [AtlasOfLivingAustralia/biocache-service#206]
The following stacktrace is on the logs of bws-b5-a although not being familiar about the qid caching system I am not sure if it is an error or not:

```
2018-04-13 11:15:00,501 [solr.client.solrj.impl.CloudSolrClient] Request to collection biocache failed due to (400) org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteSolrException: Error from server at http://aws-sc1b.ala:8983/solr/biocache: undefined field qid, retry? 0
2018-04-13 11:15:00,501 [org.ala.biocache.dao.SearchDAOImpl] query failed: qt=standard&facet=true&facet.field=taxon_name&facet.field=genus&facet.field=family&facet.field=kingdom&facet.field=common_name&facet.field=rank&facet.field=species_group&facet.field=taxonomic_issue&facet.field=type_status&facet.field=country&facet.field=state&facet.field=cl958&facet.field=cl959&facet.field=cl927&facet.field=sensitive&facet.field=geospatial_kosher&facet.field=collectors&facet.field=month&facet.field=occurrence_decade_i&facet.field=state_conservation&facet.field=basis_of_record&facet.field=multimedia&facet.field=occurrence_status&facet.field=sensitive&facet.field=assertions&facet.field=outlier_layer&facet.field=outlier_layer_count&facet.field=user_assertions&facet.field=duplicate_type&facet.field=collection_uid&facet.field=institution_uid&facet.field=data_resource_uid&f.taxon_name.facet.sort=index&f.genus.facet.sort=index&f.family.facet.sort=index&f.kingdom.facet.sort=index&f.common_name.facet.sort=index&f.rank.facet.sort=count&f.species_group.facet.sort=index&f.taxonomic_issue.facet.sort=count&f.type_status.facet.sort=index&f.country.facet.sort=index&f.state.facet.sort=index&f.cl958.facet.sort=count&f.cl959.facet.sort=count&f.cl927.facet.sort=count&f.sensitive.facet.sort=count&f.sensitive.facet.sort=count&f.geospatial_kosher.facet.sort=count&f.collectors.facet.sort=index&f.month.facet.sort=index&f.occurrence_decade_i.facet.sort=index&f.state_conservation.facet.sort=count&f.basis_of_record.facet.sort=index&f.multimedia.facet.sort=count&f.occurrence_status.facet.sort=index&f.assertions.facet.sort=count&f.outlier_layer.facet.sort=count&f.outlier_layer_count.facet.sort=count&f.user_assertions.facet.sort=count&f.duplicate_type.facet.sort=count&f.collection_uid.facet.sort=count&f.institution_uid.facet.sort=count&f.data_resource_uid.facet.sort=count&facet.mincount=1&facet.limit=10&facet.sort=count&rows=20&start=0&facet.range.other=before&facet.range.other=after&q=qid:1523581985377&facet.missing=true&sort=first_loaded_date+desc&_stateVer_=biocache-2018-04-12-23-37:55 : Error from server at http://aws-sc1b.ala:8983/solr/biocache: undefined field qid
2018-04-13 11:15:00,502 [org.ala.biocache.dao.SearchDAOImpl] Error executing query with requestParams: q=qid:1523581985377&start=0&pageSize=20&sort=first_loaded_date&dir=desc&qc=&facets=taxon_name&facets=genus&facets=family&facets=kingdom&facets=common_name&facets=rank&facets=species_group&facets=taxonomic_issue&facets=type_status&facets=country&facets=state&facets=cl958&facets=cl959&facets=cl927&facets=sensitive&facets=geospatial_kosher&facets=collectors&facets=month&facets=occurrence_decade_i&facets=state_conservation&facets=basis_of_record&facets=multimedia&facets=occurrence_status&facets=sensitive&facets=assertions&facets=outlier_layer&facets=outlier_layer_count&facets=user_assertions&facets=duplicate_type&facets=collection_uid&facets=institution_uid&facets=data_resource_uid&flimit=10&formattedQuery=qid:1523581985377&facet=true EXCEPTION: Error from server at http://aws-sc1b.ala:8983/solr/biocache: undefined field qid
org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteSolrException: Error from server at http://aws-sc1b.ala:8983/solr/biocache: undefined field qid
	at org.apache.solr.client.solrj.impl.HttpSolrClient.executeMethod(HttpSolrClient.java:610)
	at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:279)
	at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:268)
	at org.apache.solr.client.solrj.impl.LBHttpSolrClient.doRequest(LBHttpSolrClient.java:435)
	at org.apache.solr.client.solrj.impl.LBHttpSolrClient.request(LBHttpSolrClient.java:387)
	at org.apache.solr.client.solrj.impl.CloudSolrClient.sendRequest(CloudSolrClient.java:1347)
	at org.apache.solr.client.solrj.impl.CloudSolrClient.requestWithRetryOnStaleState(CloudSolrClient.java:1098)
	at org.apache.solr.client.solrj.impl.CloudSolrClient.request(CloudSolrClient.java:1037)
	at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:149)
	at org.apache.solr.client.solrj.SolrClient.query(SolrClient.java:974)
	at org.apache.solr.client.solrj.SolrClient.query(SolrClient.java:990)
	at au.org.ala.biocache.dao.SearchDAOImpl.query(SearchDAOImpl.java:3675)
	at au.org.ala.biocache.dao.SearchDAOImpl.runSolrQuery(SearchDAOImpl.java:2531)
	at au.org.ala.biocache.dao.SearchDAOImpl.findByFulltextSpatialQuery(SearchDAOImpl.java:711)
	at au.org.ala.biocache.dao.SearchDAOImpl.findByFulltextSpatialQuery(SearchDAOImpl.java:689)
	at sun.reflect.GeneratedMethodAccessor521.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:198)
	at com.sun.proxy.$Proxy31.findByFulltextSpatialQuery(Unknown Source)
	at au.org.ala.biocache.web.OccurrenceController.occurrenceSearch(OccurrenceController.java:735)
	at sun.reflect.GeneratedMethodAccessor554.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:136)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:104)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandleMethod(RequestMappingHandlerAdapter.java:743)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:672)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:82)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:933)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:867)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:951)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:842)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:624)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:827)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at au.org.ala.biocache.web.SimpleCORSFilter.doFilter(SimpleCORSFilter.java:33)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at au.org.ala.biocache.web.LoggerConfigFilter.doFilter(LoggerConfigFilter.java:26)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at com.opensymphony.sitemesh.webapp.SiteMeshFilter.obtainContent(SiteMeshFilter.java:129)
	at com.opensymphony.sitemesh.webapp.SiteMeshFilter.doFilter(SiteMeshFilter.java:77)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:88)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:106)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:221)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:169)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:956)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:436)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1078)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:316)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by javier-molina** [comment]
I could only find biocache-hubs setting the qid parameter but I don't know who in turn sets this in biocache-hubs https://github.com/AtlasOfLivingAustralia/biocache-hubs/blob/d73acdcfb33651eb1aa30e0789fb772883635e36/src/main/groovy/au/org/ala/biocache/hubs/SearchRequestParams.groovy#L138 

There is of course a DAO and a bunch of code in both biocache-service and biocache-store that deal with Qid entities, I guess we need to wait for some of the experts to shed some light on this.

**Comment by ansell** [comment]
This error has been happening in the few minutes after each switch between biocache-clustered and biocache-b4/b4, even though I have copied across the qid table a few minutes before switching. 

The error process for it could be handled better, but it only seems to trigger during that time, when people using the spatial portal have saved qids during the few minutes before the switch. Closing as not easily fixable in our current infrastructure.
